subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/CancellationSpec.groovy

```diff
@@ -72,7 +72,11 @@ latch.await()
     void buildWasCancelled(TestResultHandler resultHandler, String failureMessage = 'Could not execute build using Gradle') {
         resultHandler.assertFailedWith(BuildCancelledException)
         assert resultHandler.failure.message.startsWith(failureMessage)
-        assert resultHandler.failure.cause.message == "Build cancelled."
+
+        if (resultHandler.failure.cause != null) {
```

Comment:
Can we somehow make this decision based on the Gradle version?
IIUC, for newer Gradle versions the failure cause should never be null, right? What do we expect here?

Reply:
Done. PTAL.